Fix `falco` sample names

MartinGrignard opened the pull request:
Hi :wave:

This PR modifies the definition of the `falco` module so that the reported sample names are affected by the value of the `prefix` variable instead of the actual input file name.
To achieve this, it simply uses the same strategy as many other modules: Symlinks are now created to provide input files named after the desired `prefix`.
This makes it easier to include `falco` output in `MultiQC` reports.

In the meantime, I also updated the value of `falco` form `1.2.1` to `1.2.5`.

Commit:
Update `falco` version from `1.2.1` to `1.2.5`

Commit:
Fix `falco` sample names

Symlinks are now created as in other modules to make sure that the reported sample names are affected by the value of `prefix`. This makes it easier to use `falco` in `MultiQC` reports.

MartinGrignard committed:
Merge branch 'master' into fix/falco_sample_name

SPPearce commented:
You'll need to regenerate the test snapshots (`nf-core modules test falco --update`) and update the environment.yml to change the conda module version.

mashehu committed:
Merge branch 'master' into fix/falco_sample_name

MartinGrignard committed:
Merge branch 'master' into fix/falco_sample_name

MartinGrignard committed:
Update `falco` version from `1.2.1` to `1.2.5`

MartinGrignard committed:
Regenerate test snapshots

MartinGrignard committed:
Add snapshot test for `version` output channel

MartinGrignard committed:
Update format

MartinGrignard committed:
Fix `version` channel test

MartinGrignard committed:
Update test snapshots

Using Docker this time since the only CI test failing are those using Docker...

MartinGrignard commented:
@SPPearce do you have any idea why `nf-core modules lint` reports the following warning:

```
╭─ [!] 1 Module Test Warning ─────────────────────────────────────────────────────────────────────────╮
│              ╷                               ╷                                                      │
│ Module name  │ File path                     │ Test message                                         │
│╶─────────────┼───────────────────────────────┼─────────────────────────────────────────────────────╴│
│ falco        │ modules/nf-core/falco/main.nf │ Container versions do not match                      │
│              ╵                               ╵                                                      │
╰─────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

It is clearly related with the tests not passing since the tests fail with singularity when I generate the snapshot with Docker and vice-versa.

I used the same tag for both containers, so the reason of this error is unclear to me.

SPPearce reviewed (commented):
I'm on my phone and GitHub thinks I have an unfinished review, not sure what that actually contains. 

No, I don't see why it would be complaining, my first thought would be that the conda definition in the environment.yml would be different, but that isn't the case. They all look in sync to me

MartinGrignard commented on modules/nf-core/falco/main.nf:
```diff
@@ -1,39 +1,45 @@
-        'biocontainers/falco:1.2.1--h867801b_3' }"
+    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
+        ? 'https://depot.galaxyproject.org/singularity/falco:1.2.5--h077b44d_0'
+        : 'biocontainers/falco:1.2.5--h077b44d_0'}"
```
I tried with and without the space, with an additional new line, and nothing changes.
Actually, using the initial version of the module (`1.2.1`) yields the same result.

There is something iffy here... :thinking:

famosab replied:
Or try the seqera container:
```
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/aa/aaae9f8e330a46ac6bff3ef2c55be9315067bd42999d7305361da62b35e22208/data'
        : 'community.wave.seqera.io/library/falco:1.2.5--bd70431b39ecac2f'}"
```

famosab reviewed (commented):
Also I removed the tags.yml file: its obsolete :)

famosab commented:
To me it also seems as if the fastq files and the _fastqc_data.txt file do not have stable md5sums between conda and docker :/

SPPearce commented:
Closing due to lack of activity and merge conflicts
